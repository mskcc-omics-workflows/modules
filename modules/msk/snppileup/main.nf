process SNPPILEUP {
    tag "$meta.id"
    label 'process_medium'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/facets-suite:2.0.9':
        'docker.io/mskcc/facets-suite:2.0.9' }"

    input:

    tuple val(meta),  path(normal), path(normal_index), path(tumor),  path(tumor_index), path(additional_bams, arity: '0..*'), path(additional_bam_index, arity: '0..*')
    tuple val(meta1), path(dbsnp),  path(dbsnp_index)


    output:
    tuple val(meta), path("*.snp_pileup.gz")   , emit: pileup
    path "versions.yml"                        , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def extra_bams = additional_bams ?: ''

    """
    /usr/bin/snp-pileup \
        ${args} \
        ${dbsnp} \
        ${prefix}.snp_pileup.gz \
        ${normal} \
        ${tumor} \
        ${extra_bams}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        htslib: \$(bgzip --version | grep -oP '(?<=\\(htslib\\) ).*')
        htstools: 0.1.1
        r: \$(R --version | grep -oP '(?<=R version ).*(?=\\()')
    END_VERSIONS
    """

    stub:
    def args =   task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    echo "stub test" >> ${prefix}.snp_pileup
    gzip ${prefix}.snp_pileup
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        htslib: \$(bgzip --version | grep -oP '(?<=\\(htslib\\) ).*')
        htstools: 0.1.1
        r: \$(R --version | grep -oP '(?<=R version ).*(?=\\()')
    END_VERSIONS
    """
}