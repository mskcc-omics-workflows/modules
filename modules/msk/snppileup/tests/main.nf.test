nextflow_process {

    name "Test SNPPILEUP"
    script "../main.nf"
    process "SNPPILEUP"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "snppileup"

    test("snppileup - bam") {

        when {
            process {
                """
                    meta = [ id:'test' ] // meta map
                    input[0] = [
                                    meta,
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_normal_bam'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_normal_bam_bai'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_tumor_bam'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_tumor_bam_bai'], checkIfExists: true),
                                    [],
                                    []
                                ]
                    input[1] = [
                                    meta,
                                    file(params.test_data_mskcc['hg37']['dbsnp_138_b37_chr17_vcf_gz'], checkIfExists: true),
                                    file(params.test_data_mskcc['hg37']['dbsnp_138_b37_chr17_vcf_gz_tbi'], checkIfExists: true)
                               ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("snppileup - bam - extra bam") {

        when {
            process {
                """
                    meta = [ id:'test' ] // meta map
                    input[0] = [
                                    meta,
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_normal_bam'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_normal_bam_bai'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_tumor_bam'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_tumor_bam_bai'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_2_tumor_bam'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_2_tumor_bam_bai'], checkIfExists: true)
                                ]
                    input[1] = [
                                    meta,
                                    file(params.test_data_mskcc['hg37']['dbsnp_138_b37_chr17_vcf_gz'], checkIfExists: true),
                                    file(params.test_data_mskcc['hg37']['dbsnp_138_b37_chr17_vcf_gz_tbi'], checkIfExists: true)
                               ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("snppileup - bam - extra bam list") {

        when {
            process {
                """
                    meta = [ id:'test' ] // meta map
                    input[0] = [
                                    meta,
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_normal_bam'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_normal_bam_bai'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_tumor_bam'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_tumor_bam_bai'], checkIfExists: true),
                                    [file(params.test_data_mskcc['bam_files']['test_argos_sample_2_tumor_bam'], checkIfExists: true), file(params.test_data_mskcc['bam_files']['test_argos_sample_2_normal_bam'], checkIfExists: true) ],
                                    [file(params.test_data_mskcc['bam_files']['test_argos_sample_2_tumor_bam_bai'], checkIfExists: true), file(params.test_data_mskcc['bam_files']['test_argos_sample_2_normal_bam_bai'], checkIfExists: true) ]
                                ]
                    input[1] = [
                                    meta,
                                    file(params.test_data_mskcc['hg37']['dbsnp_138_b37_chr17_vcf_gz'], checkIfExists: true),
                                    file(params.test_data_mskcc['hg37']['dbsnp_138_b37_chr17_vcf_gz_tbi'], checkIfExists: true)
                               ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("snppileup - bam - stub") {

        options "-stub"

        when {
            process {
                """
                    meta = [ id:'test' ] // meta map
                    input[0] = [
                                    meta,
                                    file("stub_file"),
                                    file("stub_file1"),
                                    file("stub_file2"),
                                    file("stub_file3"),
                                    [],
                                    []
                                ]
                    input[1] = [
                                    meta,
                                    file("stub_file4"),
                                    file("stub_file5")
                               ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

        test("snppileup - bam with extra bam - stub") {

        options "-stub"

        when {
            process {
                """
                    meta = [ id:'test' ] // meta map
                    input[0] = [
                                    meta,
                                    file("stub_file"),
                                    file("stub_file1"),
                                    file("stub_file2"),
                                    file("stub_file3"),
                                    file("extra_bam_file"),
                                    file("extra_bam_bai_file")
                                ]
                    input[1] = [
                                    meta,
                                    file("stub_file4"),
                                    file("stub_file5")
                               ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("snppileup - bam with extra bam list - stub") {

        options "-stub"

        when {
            process {
                """
                    meta = [ id:'test' ] // meta map
                    input[0] = [
                                    meta,
                                    file("stub_file"),
                                    file("stub_file1"),
                                    file("stub_file2"),
                                    file("stub_file3"),
                                    [file("extra_bam_file"), file("extra_bam_file_1")],
                                    [file("extra_bam_bai_file"), file("extra_bam_bai_file_1")]
                                ]
                    input[1] = [
                                    meta,
                                    file("stub_file4"),
                                    file("stub_file5")
                               ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
