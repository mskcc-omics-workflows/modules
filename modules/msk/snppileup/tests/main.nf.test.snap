{
    "snppileup - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,3d77e3e110a219d3495d44e243450313"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ],
                "pileup": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,3d77e3e110a219d3495d44e243450313"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ]
            }
        ],
        "timestamp": "2024-02-26T11:34:57.371873"
    },
    "snppileup - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,deac57ec8a27969028af701840d2b0ee"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ],
                "pileup": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,deac57ec8a27969028af701840d2b0ee"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ]
            }
        ],
        "timestamp": "2024-02-26T11:33:58.146"
    },
    "snppileup - bam with extra bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,3d77e3e110a219d3495d44e243450313"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ],
                "pileup": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,3d77e3e110a219d3495d44e243450313"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ]
            }
        ],
        "timestamp": "2024-02-26T11:35:02.583042"
    },
    "snppileup - bam - extra bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,28e74d5b98f0f91960d09a229c755b70"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ],
                "pileup": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,28e74d5b98f0f91960d09a229c755b70"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ]
            }
        ],
        "timestamp": "2024-02-26T11:34:22.986163"
    },
    "snppileup - bam - extra bam list": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,5cd252c71059426451f386ff386b8534"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ],
                "pileup": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,5cd252c71059426451f386ff386b8534"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ]
            }
        ],
        "timestamp": "2024-02-26T11:34:51.747062"
    },
    "snppileup - bam with extra bam list - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,3d77e3e110a219d3495d44e243450313"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ],
                "pileup": [
                    [
                        {
                            "id": "test"
                        },
                        "test.snp_pileup.gz:md5,3d77e3e110a219d3495d44e243450313"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0ca251848ee099aebdade397bedf463f"
                ]
            }
        ],
        "timestamp": "2024-02-26T11:35:07.596619"
    }
}